### DISCUSSION TITLE
When Jumping 1 second, the window beneath SMPlayer displays

### DISCUSSION THREAD

**Discussion by smp-marco** [smplayer-dev/smplayer#568] [Q&A]
Version: 22.7.0 (revision 10091) (64-bit)
Using Qt 5.15.2 (compiled with Qt 5.15.2)
Using MPV 0.34.0

When Jumping 1 second, the window beneath SMPlayer displays momentarily.
If the jump button is pressed repeatedly, a mouse click can register on the window beneath, bringing that window to the foreground.

Perhaps not a major issue.
However, it is noticeable, and slightly distracting.

It does give the impression that something is not quite right.
:)

**Comment by smplayer-dev** [comment]
I think I've never seen something like this. Just in case sure the "stay on top" option is disabled.

**Reply by smp-marco** [reply]
This is an interesting one :D
It's the "On Top while playing" Option that is causing the problem.

I first checked "Always" and it fixed the issue.

My immediate thoughts were that during the jump, the window lost focus because the vid wasn't playing.
However, I then checked "Never", and this also fixed the issue.
This blew my theory.

Next, I concentrated on the whole window.
The whole window momentarily, was partially disappearing.
It was Full Screen, so I dropped it to Resizeable, and the entire browser GUI was glitching.
ie. the tab titles were changing colour, as they do, when the browser window gains focus.

I tested with just the windows desktop.
The desktop was stable.

The good news was that when resizeable, the window misbehaviour lasted longer, so I could see that the window was momentarily disappearing.

Consequently, the SMP window disappears so focus immediately switches to the next layer down.

Hahah!
Just to drop one more element into the mix.
While penning this text, I was going back and forth checking what I was seeing, and when I wanted to check the browser tab text colour
... It was stable at first!
When I continued jumping, but faster, the glitch returned.

So I've just checked this now.
It's not the speed of the jump clicks.

When I switch focus to the SMP window, and start to click jump
... the window momentarily disappears, but the browser window might not recognise it has focus, or it might.
However, after clicking a few times, it always recognises that it has focus (for those few milliseconds).

Nice eh?
This, from choosing to stay on top while playing.
It's like the "flapping butterfly wing" effect, that resulted in a misbehaving window.
Amazing :)

I just hope that you can reproduce the effect.
Here's my system details:

Device name	DESKTOP-VAUNPT0
Processor	Intel(R) Core(TM) i5-4590 CPU @ 3.30GHz   3.30 GHz
Installed RAM	8.00 GB
Device ID	323D0A44-9007-44C7-8B98-5131333431CE
Product ID	00331-10000-00001-AA584
System type	64-bit operating system, x64-based processor
Pen and touch	No pen or touch input is available for this display

Edition	Windows 10 Pro
Version	21H2
Installed on	‎Fri ‎19/‎03/‎21
OS build	19044.2130
Experience	Windows Feature Experience Pack 120.2212.4180.0
